x/incentive/keeper/keeper.go

```diff
@@ -39,10 +45,267 @@ func NewKeeper(
+		cmk:        ck,
+		stk:        sk,
+		tci:        &types.TotalCommitmentInfo{},
 	}
 }
 
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+// Update total commitment info
+func (k Keeper) UpdateTotalCommitmentInfo(ctx sdk.Context, epochIdentifier string) {
+	// Fetch total staked Elys amount again
+	k.tci.TotalElysBonded = k.stk.TotalBondedTokens(ctx)
+	// Initialize with amount zero
+	k.tci.TotalEdenCommitted = sdk.ZeroInt()
+	k.tci.TotalEdenBoostCommitted = sdk.ZeroInt()
+
+	// Iterate to calculate total Eden and Eden boost committed
+	k.cmk.IterateCommitments(ctx, epochIdentifier, func(commitments ctypes.Commitments) bool {
+		committedTokens := commitments.CommittedTokens
+		// Sum Eden and Eden boost committed
+		for _, c := range committedTokens {
+			// Eden
+			if c.Denom == types.Eden {
+				k.tci.TotalEdenCommitted = k.tci.TotalEdenCommitted.Add(c.Amount)
+			}
+
+			// Eden boost
+			if c.Denom == types.EdenB {
+				k.tci.TotalEdenBoostCommitted = k.tci.TotalEdenCommitted.Add(c.Amount)
+			}
+		}
+
+		return false
+	})
+}
+
+// Calculate total share of staking
+func (k Keeper) CalculateTotalShareOfStaking(amount sdk.Int) sdk.Dec {
+	// Total statked = Elys staked + Eden Committed + Eden boost Committed
+	totalStaked := k.tci.TotalElysBonded.Add(k.tci.TotalEdenCommitted).Add(k.tci.TotalEdenBoostCommitted)
+
+	// Share = Amount / Total Staked
+	return sdk.NewDecFromInt(amount).QuoInt(totalStaked)
+}
+
+// Calculate the delegated amount
+func (k Keeper) CalculateDelegatedAmount(ctx sdk.Context, delegator string) sdk.Int {
+	// Derivate bech32 based delegator address
+	delAdr, err := sdk.AccAddressFromBech32(delegator)
+	if err != nil {
+		return sdk.ZeroInt()
+	}
+
+	// Get elys delegation for creator address
+	delegatedAmt := sdk.ZeroInt()
+
+	// Iterate all delegations for the specified delegator
+	k.stk.IterateDelegations(
+		ctx, delAdr,
+		func(_ int64, del stakingtypes.DelegationI) (stop bool) {
+			// Get validator address
+			valAddr := del.GetValidatorAddr()
+			// Get validator
+			val := k.stk.Validator(ctx, valAddr)
+
+			/****************************************************************************************/
+			// --------------------(Cosmos SDK staking module implementation)------------------------
+			// del.Shares = val.GetDelegatorShares().MulInt(amt).QuoInt(val.GetTokens())
+			// amt = del.Shares * val.GetTokens / val.GetDelegatorShares()
+			/****************************************************************************************/
+			amt := del.GetShares().MulInt(val.GetTokens()).Quo(val.GetDelegatorShares()).TruncateInt()
+
+			// Sum the individual delegation
+			delegatedAmt.Add(amt)
+			return false
+		},
+	)
+
+	return delegatedAmt
+}
+
+// Find out active incentive params
+func (k Keeper) FindProperIncentiveParm(ctx sdk.Context, epochIdentifier string) (bool, types.IncentiveInfo, types.IncentiveInfo) {
```

Comment:
Can you update the name to full version? `FindProperIncentiveParam` or `GetProperIncentiveParam`

Reply:
The name dooes not make sense I think.

Reply:
Ok